jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/util/KubernetesResourceUtil.java

```diff
@@ -650,76 +650,115 @@ public static String mergePodSpec(PodSpecBuilder builder, PodSpec defaultPodSpec
         int size = defaultContainers.size();
         if (size > 0) {
             if (containers == null || containers.isEmpty()) {
+                defaultApplicationContainerName = defaultContainers.get(0).getName();
```

Comment:
Do we always the guarantee that the 0th container is the app container? I think we should add a validation here to keep this logic robust.

Please correct me if I'm wrong.

Reply:
I think you have a point here. During zero-configuration mode, JKube adds only one container(that's why I assumed it would always work). But if you use XML configuration for ImageConfig, you can add as many containers as you wish and there is no way to specify which one is main application container.

Reply:
exactly

Reply:
With all the code changes, I just realized that this diverges a lot from the initial [proposed solution](https://github.com/eclipse/jkube/issues/341#issuecomment-669915655) (I thought that code was only moved around).

Could we stick to the initial plan (implement simple solution and add the the tests)? This class needs a deep refactor, but with the 1.0 release ahead, it's not the right time to do this.

Reply:
After a deeper review, the string returned by this function is absolutely useless and only used as a flag in:

https://github.com/eclipse/jkube/blob/89f36498ab7e2dcc3d7d233e0ae919f4f05fe451/jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/ControllerViaPluginConfigurationEnricher.java#L123-L128

 It doesn't really matter if the name is the one for the first container, the second, or anything else `"random"`. 

We really need to refactor all of the related methods after the 1.0.0 release